services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/update/AddUpdateAction.java

```diff
@@ -0,0 +1,208 @@
+ * <p>
+ * Example:-
+ * <pre>
+ * {@code
+ * AddUpdateAction addAction = AddUpdateAction.builder()
+ *                                            .path("#a")
+ *                                            .value(":b")
+ *                                            .putExpressionName("#a", "attributeA")
+ *                                            .putExpressionValue(":b", myAttributeValue)
+ *                                            .build();
+ * }
+ * </pre>
+ */
+@SdkPublicApi
+public final class AddUpdateAction implements UpdateAction {
+
+    private final String path;
+    private final String value;
+    private final Map<String, String> expressionNames;
+    private final Map<String, AttributeValue> expressionValues;
+
+    private AddUpdateAction(Builder builder) {
+        this.path = Validate.paramNotNull(builder.path, "path");
+        this.value = Validate.paramNotNull(builder.value, "value");
+        this.expressionValues = unmodifiableMap(Validate.paramNotNull(builder.expressionValues, "expressionValues"));
+        this.expressionNames = unmodifiableMap(builder.expressionNames != null ? builder.expressionNames : new HashMap<>());
+    }
+
+    /**
+     * Constructs a new builder for {@link AddUpdateAction}.
+     *
+     * @return a new builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public String path() {
+        return path;
+    }
+
+    public String value() {
+        return value;
+    }
+
+    public Map<String, String> expressionNames() {
+        return expressionNames;
+    }
+
+    public Map<String, AttributeValue> expressionValues() {
+        return expressionValues;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        AddUpdateAction that = (AddUpdateAction) o;
+
+        if (path != null ? ! path.equals(that.path) : that.path != null) {
+            return false;
+        }
+        if (value != null ? ! value.equals(that.value) : that.value != null) {
+            return false;
+        }
+        if (expressionValues != null ? ! expressionValues.equals(that.expressionValues) :
+            that.expressionValues != null) {
+            return false;
+        }
+        return expressionNames != null ? expressionNames.equals(that.expressionNames) : that.expressionNames == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = path != null ? path.hashCode() : 0;
+        result = 31 * result + (value != null ? value.hashCode() : 0);
+        result = 31 * result + (expressionValues != null ? expressionValues.hashCode() : 0);
+        result = 31 * result + (expressionNames != null ? expressionNames.hashCode() : 0);
+        return result;
+    }
+
+    /**
+     * A builder for {@link AddUpdateAction}
+     */
+    public static final class Builder {
```

Comment:
Nit: Can we implement CopyableBuilder and ToCopyableBuilder on these classes at some point?

Reply:
Will add to task list.